frontend/pkg/frontend/frontend.go

```diff
@@ -211,7 +211,39 @@ func (f *Frontend) ArmResourceListBySubscription(writer http.ResponseWriter, req
 
 	f.logger.Info(fmt.Sprintf("%s: ArmResourceListBySubscription", versionedInterface))
 
-	writer.WriteHeader(http.StatusOK)
+	subscriptionId := request.PathValue(PathSegmentSubscriptionID)
+
+	query := fmt.Sprintf("azure.subscription_id='%s'", subscriptionId)
+	clustersListResponse, err := f.conn.ClustersMgmt().V1().Clusters().List().Search(query).Send()
+	if err != nil {
+		f.logger.Error(err.Error())
+		arm.WriteInternalServerError(writer)
+		return
+	}
+
+	var hcpCluster *api.HCPOpenShiftCluster
+	clusters := clustersListResponse.Items().Slice()
+	for _, cluster := range clusters {
+		hcpCluster, err = f.ConvertCStoHCPOpenShiftClusterbySubscription(cluster)
+		if err != nil {
+			f.logger.Error(err.Error())
+			arm.WriteInternalServerError(writer)
+			return
+		}
+
+		versionedResource := versionedInterface.NewHCPOpenShiftCluster(hcpCluster)
+		resp, err := json.MarshalIndent(versionedResource, "", "    ")
```

Comment:
What's the purpose of `json.MarshalIndent`? I believe this is unique compared to the other `ArmResource*` functions, so asking out of curiosity

Reply:
That indents the output by line instead of a big paragraph

Reply:
We should centralize JSON marshalling in the frontend for consistency.  Like, add a `MarshalJSON` function instead of calling `json.Marshal` or `json.MarshalIndent` directly.  Not sure if there's an ARM requirement for or against pretty-printing response bodies, but either way it should be determined in one place.